lib/sidekiq/flavors/active_job.rb

```diff
@@ -0,0 +1,125 @@
+module Sidekiq
+  module Flavors
+    module ActiveJob
+      ACTIVE_JOB_PREFIX = "_aj_"
+      GLOBALID_KEY = "_aj_globalid"
+      SYMBOL_KEYS_KEY = "_aj_symbol_keys"
+      RUBY2_KEYWORDS_KEY = "_aj_ruby2_keywords"
+
+      def self.validate_job(job)
+        # TODO
+      end
+
+      def self.flavor_job(job)
+        if job["class"] == "ActiveJob::QueueAdapters::SidekiqAdapter::JobWrapper" || job["class"] == "Sidekiq::ActiveJob::Wrapper"
+          # Don't flavor jobs that came from ActiveJob itself.
+          job
+        elsif job["_f"] == "aj"
+          # Don't flavor jobs that are already flavored.
+          job
+        else
+          job["_f"] = "aj"
+          job["args"] = ::ActiveJob::Arguments.serialize(job["args"])
+          job
+        end
+      end
+
+      def self.flavor_hash(hash)
+        {
+          "_f" => "aj",
+          "args" => ::ActiveJob::Arguments.serialize([hash])
+        }
+      end
+
+      def self.valid_for_unflavor?(type:, item:)
+        item["_f"] == "aj"
+      end
+
+      def self.unflavor_job(job)
+        job["args"] = ::ActiveJob::Arguments.deserialize(job["args"])
+        job.delete("_f")
+        job
+      end
+
+      def self.unflavor_hash(hash)
+        ::ActiveJob::Arguments.deserialize(hash["args"]).first
+      end
+
+      def self.valid_for_display?(type:, item:)
+        case type
+        when :job
+          job_record = item
+          job_record.klass == "ActiveJob::QueueAdapters::SidekiqAdapter::JobWrapper" || job_record.klass == "Sidekiq::ActiveJob::Wrapper" || job_record["_f"] == "aj"
+        when :hash
+          hash = item
+          hash["_f"] == "aj"
+        end
+      end
+
+      def self.display_args(job_record)
```

Comment:
@mperham How do we want to handle the case where a user has multiple ruby services sharing a sidekiq server? It's possible that only some of them are Rails-based and the server might not have this flavor loaded. It's a bit of an edge case, do we care?

Reply:
I'm not sure that's a mode of operation we want to officially support. Just fall back to `to_s`?